### ISSUE TITLE
Add 'from' setting to /mail endpoint

### ISSUE THREAD

**Issue by stefanvangastel** [directus/api#841]
# Feature Request

## Steps to Reproduce

1. Configure mail in `api.php` or any project `api.php` file:
```php
'default' => [
  'transport' => 'sendmail',
  'from' => 'noreply@mydomain.com'
]
```
2. Mail some recipient (user@example.com) using the `/mail` endpoint and provide access via token:
```
POST /_/mail?access_token=adminUserSecretToken

{
  "to": [
    "user@example.com"
  ],
  "subject": "New Password",
  "body": "Hello <b>{{name}}</b>, this is your new password: {{password}}.",
  "type": "html",
  "data": {
    "name": "John Doe",
    "password": "secret"
  }
}
```
3. As user@example.com check your email and the sender address.

## Expected Behavior

The sender address is: `noreply@mydomain.com`

## Actual Behavior

The sender address is the email address of the (admin) user corresponding to the used access_token: `admin_user_corresponding_to_token@email.address`

## Other Context & Screenshots

Nope.

## Technical Details

- Device: Desktop, Chrome latest
- OS: MacOs latest
- Web Server: Apache 2.4
- PHP Version: 7.2
- Database: 5.6
- Install Method: Cloned master branch

**Comment by theharshin** [comment]
Thank you for pointing this out @stefanvangastel. We'll be looking at this issue soon 🙂

**Comment by bjgajjar** [comment]
Hello, this is working as expected. What you're trying to achieve can be added as enhancement though. Let me explain,

The api.config file has mail settings which are being used when the system sends an email for activities like:
1. New registration
2. Forgot passwords etc.

The /mail endpoint is open for all the users. It identifies the user from access token & sets the email of the user as sender. Now there may be some cases where you might need to tweak this behaviour. To do that we can add an extra parameter use_default_email which will user the email address from configuration.

This may create a situation where other user can also pass this parameter and represent them as system email. So we can add check that if token is of an admin then only use_default_email will come to a picture!

**Comment by stefanvangastel** [comment]
@bjgajjar That explains a lot. Thanks. I understand this point of view and the suggested feature would be very helpful for my (and I expect other) use-case(s).

**Comment by benhaynes** [comment]
Makes sense, thanks Binal!

Let's shift this ticket to a feature request for now and we can look into adding it later. For now the priority is bug fixes and stability.

**Comment by bjgajjar** [comment]
Fixed in #911
